File: problem_002.c
// -*- compile-command: "gcc -std=c99 -o problem_002 problem_002.c ceuler.c -Wall -lm" -*-
// Copyright (c) 2014 Michael Caldwell
#include <stdio.h>
#include "ceuler.h"

int main()
{
    printf("Project Euler - Problem 2:\n"
           "By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.\n\n");

    // Begin time tracking
    struct timeval start;
    gettimeofday(&start, NULL);

    const int limit = 4000000;
    int previousNum = 0, nextNum = 0, currentNum = 1;
    int total = 0;

    while (currentNum < limit)
    {
        nextNum = previousNum + currentNum;
        previousNum = currentNum;
        currentNum = nextNum;
        if ( currentNum % 2 == 0)
            total += currentNum;
    }

    printf("Sum: %d\n", total);
    printElapsedTime(start);

    return 0;
}

File: problem_002.kt
// -*- compile-command: "kotlinc problem_002.kt -include-runtime -d problem_002.jar" -*-
// Copyright (c) 2017 Michael Caldwell

fun main(args: Array<String>) {
        println("Project Euler - Problem 2:")
        println("By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.\n")

        val limit: Int = 4000000
        var n: Int = 1
        var a: Int = 0
        var sum: Int = 0

        while (n < limit) {
                val t: Int = n
                n += a
                a = t

                if (n % 2 == 0) {
                        sum += n
                }
        }

        println(sum)
}
